### PR TITLE
Added /consensus_params endpoint 

### LINKED ISSUE #2311: expose consensus params over RPC
Like `/validators?height=X`, we should be able to query for `/consensus_params?height=X`

### PR DESCRIPTION
**Pull request by scriptonist**
* [ ] Updated all relevant documentation in docs
* [x] Updated all code comments where relevant
* [x] Wrote tests
* [ ] Updated CHANGELOG_PENDING.md

Fixes #2311

### CODE REVIEW

**Comment by scriptonist** [comment]
I would really appreciate some early feedback

File: rpc/core/types/responses.go [hunk 1]
```diff
@@ -125,6 +125,12 @@ type ResultValidators struct {
 	Validators  []*types.Validator `json:"validators"`
 }
 
+// ConsensusParams for given height
+type ResultConsensusParams struct {
+	BlockHeight     int64                  `json:"block_height"`
+	ConsensusParams *types.ConsensusParams `json:"consensus_params"`
```
Comment (xla) [inline rpc/core/types/responses.go]: Minor nit, should not be a pointer as it is non-optional.

**Comment by melekes** [comment]
This also needs a changelog entry (`CHANGELOG_PENDING.md`).

**Comment by xla** [comment]
@zramsay Any docs that need to be updated with this new endpoint being introduced?

File: rpc/core/consensus.go [hunk 2]
```diff
@@ -261,3 +261,49 @@ func ConsensusState() (*ctypes.ResultConsensusState, error) {
...
+// }
+// ```
+func ConsensusParams(heightPtr *int64) (*ctypes.ResultConsensusParams, error) {
+	height := consensusState.GetState().LastBlockHeight + 1
+	height, err := getHeight(height, heightPtr)
+	if err != nil {
+		return nil, err
+	}
+
+	consensusparams, err := sm.LoadConsensusParams(stateDB, height)
```
Comment (ebuchman) [inline rpc/core/consensus.go]: `consensusParams`

Comment (scriptonist) [reply rpc/core/consensus.go]: @ebuchman What should I fill in the changelog entry ? can you help me with that?

Comment (scriptonist) [reply rpc/core/consensus.go]: Should I open a new PR for incorporating these changes ?

Comment (ebuchman) [reply rpc/core/consensus.go]: no worries we'll cover it! thanks again for the contribution!

for reference, for the next PR, add a note about your change in the CHANGELOG_PENDING.md file: https://github.com/tendermint/tendermint/blob/develop/CHANGELOG_PENDING.md

Be sure to follow the formats used there :)

Comment (scriptonist) [reply rpc/core/consensus.go]: Sure thanks :smile:

**Pull request merged**